pkgs/_macro_host/lib/src/package_config.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:dart_model/dart_model.dart';
+import 'package:package_config/package_config.dart' as package_config;
+
+/// Reads a package config to determine information about macros.
+class PackageConfig {
```

Comment:
Lets also call this something other than PackageConfig. Really, it is something that reads a package config and extracts information about macros from it.

Reply:
I guess: MacroPackageConfig :) done.